src/representations/representation_manager_spherical_expansion.hh

```diff
@@ -0,0 +1,535 @@
+
+      this->hypers = hypers;
+      if (this->gaussian_sigma_str.compare("Constant") == 0) {
+        this->gaussian_sigma_type = internal::GaussianSigmaType::Constant;
+      } else {
+        throw std::logic_error("Requested Gaussian sigma type \'"
+                               + gaussian_sigma_str
+                               + "\' has not been implemented.  Must be one of"
+                               + ": \'Constant\'.");
+      }
+    }
+
+    /**
+     * Set the hyperparameters of this descriptor from a json string
+     *
+     * @param hypers json string containing the options and hyperparameters
+     *               The string will be parsed into a json container.
+     *
+     * @throw logic_error if an invalid option or combination of options is
+     *                    specified in the string
+     *
+     * @throw json.exception.parse_error if the string is not valid json
+     */
+    void set_hyperparameters(const std::string& hypers) {
+      hypers_t hypers_structure = json::parse(hypers);
+      this->set_hyperparameters(hypers_structure);
+    }
+
+    /**
+     * Construct a new RepresentationManager using a hyperparameters container
+     *
+     * @param hypers container (usually parsed from json) for the options and
+     *               hyperparameters
+     *
+     * @throw logic_error if an invalid option or combination of options is
+     *                    specified in the container
+     */
+    RepresentationManagerSphericalExpansion(Manager_t &sm,
+                                            const hypers_t& hyper)
+        :structure_manager{sm}, soap_vectors{sm}
+    {
+      this->set_hyperparameters(hyper);
+    }
+
+    //! Copy constructor
+    RepresentationManagerSphericalExpansion(
+      const RepresentationManagerSphericalExpansion &other) = delete;
+
+    //! Move constructor
+    RepresentationManagerSphericalExpansion(
+      RepresentationManagerSphericalExpansion &&other) = default;
+
+    //! Destructor
+    virtual ~RepresentationManagerSphericalExpansion()  = default;
+
+    //! Copy assignment operator
+    RepresentationManagerSphericalExpansion& operator=(
+      const RepresentationManagerSphericalExpansion &other) = delete;
+
+    //! Move assignment operator
+    RepresentationManagerSphericalExpansion& operator=(
+      RepresentationManagerSphericalExpansion && other) = default;
+
+    //! compute representation
+    void compute();
+
+    template<internal::GaussianSigmaType SigmaType>
+    void compute_by_gaussian_sigma();
+
+    //! Precompute radial Gaussian widths (NB specific to basis!)
+    void precompute_radial_sigmas();
+
+    //! Precompute radial orthogonalization matrix (NB specific to basis!)
+    void precompute_radial_overlap();
+
+    //! Precompute everything that doesn't depend on the structure
+    void precompute();
+
+    //! getter for the representation
+    template <size_t Order, size_t Layer>
+    Eigen::Map<Eigen::MatrixXd> get_soap_vector(
```

Comment:
This was chosen to be consistent (at least in spirit) with the Sorted Coulomb representation, but the naming might have moved on since. We can discuss interface conventions at the next meeting.